src/astbuilder.nit

```diff
@@ -934,3 +934,205 @@ redef class AAnnotation
 		_n_args = n_args
 	end
 end
+
+redef class MEntity
+	# Return a ast representation of self model entity.
+	# This method simplifies the creation of the ast based on the information available in the model.
+	fun create_ast_representation(astbuilder: nullable ASTBuilder): ANode is abstract
+end
+
+redef class MPropDef
+	redef fun create_ast_representation(astbuilder: nullable ASTBuilder): APropdef is abstract
+end
+
+redef class MClassDef
+	redef fun create_ast_representation(astbuilder: nullable ASTBuilder): AStdClassdef do
+		if astbuilder == null then astbuilder = new ASTBuilder(mmodule)
+		var n_propdefs = new Array[APropdef]
+		for mpropdef in self.mpropdefs do
+			n_propdefs.add(mpropdef.create_ast_representation(astbuilder))
+		end
+		var n_formaldefs = new Array[AFormaldef]
```

Comment:
RIP generics I guess?

Reply:
Good catch, I have fixed that problem now.